### DISCUSSION TITLE
Does bun has any docker image available?

### DISCUSSION THREAD

**Discussion by timam** [oven-sh/bun#1040] [Q&A] [answered]
I am looking for docker image for bun. I searched if there is any docker image available for bun, couldn't find any.

**Comment by Bellisario** [comment, marked as answer]
Yes, there is a Docker image available for Bun. You can find it on the [installation section](https://bun.sh/docs/installation) of Bun docs...

Here it is:
> Docker
> 
> ```bash
> docker pull oven/bun
> ```
> ```bash
> docker run --rm --init --ulimit memlock=-1:-1 oven/bun
> ```

**Reply by timam** [reply]
I have created react app with ` bun create react`
```
```
Here is the content of my Dockerfile
```
FROM jarredsumner/bun:edge
COPY . .
EXPOSE 3000
ENTRYPOINT ["bun",  "dev"]
```
Next I am building docker Image
```
$  docker build . -t reactbun
[+] Building 3.0s (8/8) FINISHED                                                
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 36B                                        0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for docker.io/jarredsumner/bun:edge           2.9s
 => [auth] jarredsumner/bun:pull token for registry-1.docker.io            0.0s
 => [internal] load build context                                          0.1s
 => => transferring context: 37.28kB                                       0.0s
 => [1/2] FROM docker.io/jarredsumner/bun:edge@sha256:8eb17260732aa04fe4d  0.0s
 => CACHED [2/2] COPY . .                                                  0.0s
 => exporting to image                                                     0.0s
 => => exporting layers                                                    0.0s
 => => writing image sha256:216736b059d26dab1f72a12307f3dc58d1f11d52a6380  0.0s
 => => naming to docker.io/library/reactbun                                0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
```

Now if I try to run this docker image, I am getting following error
```
$ docker run --rm --init --ulimit memlock=-1:-1 reactbun -p 3000:3000
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested

error: InvalidPort
```

**Reply by Bellisario** [reply]
Seems you're using an ARM processor, for now, Bun doesn't support it (#75) without Docker, but you're using Docker (and should work) :face_with_spiral_eyes:...
The strange thing is that @Jarred-Sumner commented [here](https://github.com/oven-sh/bun/issues/75#issuecomment-1004645261) Docker image should work on ARM arch, too, so where is the problem?
Maybe I'm not understanding something... :disappointed:

**Reply by muuvmuuv** [reply]
Try it just now but still no ARM docker target available

**Comment by NWylynko** [comment]
I built this today, every bun version from 0.1.1 to 0.1.10 https://hub.docker.com/r/nwylynko/bun/tags I made a github action to check every 24 hours for a new release and will build a new docker image.

**Reply by NWylynko** [reply]
would love to make arm version but not sure how to go about doing that

**Comment by tonyhb** [comment]
Yeah, Bun right now only has x86 docker images available;  the last image push was a month ago.

I think we need to work on Image Manifest support.  This lets a single image manifest have multiple image architectures supported — you pull one tag `:edge` and it'll automatically choose the right platform.

All we really need is `linux` // [`arm64`, `amd64`] because D4M uses linux under the hood.

Without this, Bun is dead in the water on Docker.  A `bun install` takes over 10 minutes to run with... 5 packages.

**Reply by Jarred-Sumner** [reply]
Just pushed 0.1.10 for linux/amd64

linux/arm64 needs to not use alpine or have a glibc compat build for linux/arm64

**Comment by duongdam** [comment]
https://gist.github.com/duongdam/7fbcdf02c38b58a61d3a18c5b9732136
Try this!

**Reply by ducphd2** [reply]
many thanks
